File: app/src/main/java/org/evcs/android/model/PricingDetail.java
package org.evcs.android.model;

import androidx.annotation.Nullable;

import java.io.Serializable;

public class PricingDetail implements Serializable {
    public Float priceKwh;
    private Float initialFee;
    private Float occupancyFlatFee;
    private Integer bufferTime;
//    int firstHour;
//    int firstHourPrice;
    public Integer thereafterPrice;
//    int regularMinimumCost;
    public boolean showFreeChargingCode;
    public String freeChargingCode;

    //TODO: chequear que pasa si es null
    public String printPriceKwh() {
        return String.format("%.2f USD/kWh", priceKwh);
    }

    public @Nullable String printConnectionFee() {
        if (initialFee == null) return null;
        if (initialFee == 0f) return "Free";
        return initialFee.toString();
    }

    public @Nullable String printOccupancyFee() {
        if (occupancyFlatFee == null) return null;
        return String.format("%.2f USD/charge", occupancyFlatFee);
    }

    public @Nullable String printBufferTime() {
        if (bufferTime == null) return null;
        return String.format("%d mins", bufferTime);
    }
}

File: app/src/main/java/org/evcs/android/model/PricingDetail.kt
package org.evcs.android.model

import java.io.Serializable

class PricingDetail : Serializable {
    var priceKwh: Float? = null
    private val initialFee: Float? = null
    private val occupancyFlatFee: Float? = null
    private val bufferTime: Int? = null

    //    int firstHour;
    //    int firstHourPrice;
    var thereafterPrice: Int? = null

    //    int regularMinimumCost;
    var showFreeChargingCode = false
    var freeChargingCode: String? = null

    //TODO: chequear que pasa si es null
    fun printPriceKwh(): String {
        return String.format("%.2f USD/kWh", priceKwh)
    }

    fun printConnectionFee(): String? {
        if (initialFee == null) return null
        return if (initialFee == 0f) "Free" else initialFee.toString()
    }

    fun printOccupancyFee(): String? {
        return if (occupancyFlatFee == null) null else String.format("%.2f USD/charge", occupancyFlatFee)
    }

    fun printBufferTime(): String? {
        return if (bufferTime == null) null else String.format("%d mins", bufferTime)
    }
}